pythonfmu/pythonfmu-export/cpp/PySlaveInstance.cpp

```diff
@@ -418,7 +418,9 @@ void PySlaveInstance::handle_py_exception(const std::string& what, PyGILState_ST
         oss << what << "\n";
         if (pExcValue != nullptr) {
             PyObject* pRepr = PyObject_Repr(pExcValue);
-            oss << PyBytes_AsString(PyUnicode_AsEncodedString(pRepr, "utf-8", nullptr));
+            PyObject* pyStr = PyUnicode_AsEncodedString(pRepr, "utf-8", nullptr);
+            oss << PyBytes_AsString(pyStr);
+            Py_DECREF(pyStr);
```

Comment:
I wonder if `oss.str()` work after this?

Reply:
The new test (https://github.com/NTNU-IHB/PythonFMU/blob/60a3ffce139ceb024e773bf840b2224faa912f4b/pythonfmu/tests/test_integration.py#L302) passed. So this seems ok.

Running the test locally with `pytest pythonfmu -sv` to see the output. I got:

```
../pythonfmu/tests/test_integration.py::test_integration_throw_py_error [FATAL] Fatal py exception encountered: [doStep] PyObject_CallMethod
RuntimeError()
PASSED
```